docs/source/loading.rst

```diff
@@ -0,0 +1,304 @@
+Load
+====
+
+You have already seen how you can load a dataset from the Datasets Hub. But datasets are stored in a variety of places, and sometimes you won't find the one you want on the Hub. The datasets can be on your local machine, in a Github repository, and in data structures like Python dictionaries and Pandas DataFrames. Wherever your dataset may be stored, Datasets provides a way for you to load it and use it for training.
+
+This guides shows you how to load a dataset from:
+
+* local files
```

Comment:
We will need to update this part when https://github.com/huggingface/datasets/pull/2662 is merged. This PR added some documentation on how to load a dataset from your own dataset repository on the Hugging Face Hub, or from any remote URLs. You can already cherry-pick the changes from this PR :)

We probably want a special focus on how to load a dataset from a dataset repository on the Hub

Reply:
Added how-to for loading a dataset from a repo on the Hub (put it first on the list for more emphasis :) ), and a brief explanation in the loading datasets explanation. Feel free to add more details!